.github/workflows/dockerimage-build.yml

```diff
@@ -0,0 +1,16 @@
+name: Docker Image CI
+
+on: [push]
+
+jobs:
+
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Build the Docker image
+      run: | 
+        docker build . --file docker/Dockerfile --tag my-image-name:$(date +%s)
+        docker run --name testrun my-image-name:$(date +%s)
```

Comment:
need to create `$(date +%s)` once or it will change between build and run.
Also (nitpicking) maybe split steps as one build and one run